app/uk/gov/hmrc/tai/connectors/BaseConnector.scala

```diff
@@ -43,11 +44,13 @@ abstract class BaseConnector(auditor: Auditor, metrics: Metrics, httpClient: Htt
     npsVersion
   }
 
-  def extraNpsHeaders(hc: HeaderCarrier, version: Int, txId: String): HeaderCarrier =
-    hc.withExtraHeaders("ETag" -> version.toString, "X-TXID" -> txId, "Gov-Uk-Originator-Id" -> originatorId)
-
   def basicNpsHeaders(hc: HeaderCarrier): HeaderCarrier =
-    hc.withExtraHeaders("Gov-Uk-Originator-Id" -> originatorId)
+    hc.withExtraHeaders(
+      "Gov-Uk-Originator-Id" -> originatorId,
+      HeaderNames.xSessionId -> hc.sessionId.fold("-")(_.value),
+      HeaderNames.xRequestId -> hc.requestId.fold("-")(_.value),
+      "CorrelationId"        -> UUID.randomUUID().toString
+    )
 
   def getFromNps[A](url: String, api: APITypes)(implicit hc: HeaderCarrier, formats: Format[A]): Future[(A, Int)] = {
```

Comment:
We should pass through the explicit headers here similar to the change made to the post call below

`def postToNps[A](url: String, api: APITypes, postData: A, headers: Seq[(String, String)])(`

Reply:
Done